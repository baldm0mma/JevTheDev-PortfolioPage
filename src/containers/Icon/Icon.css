.icon {
  width: 50px;
  height: auto;
  transition: transform 0.2s ease;
}

.icon:hover {
  transform: scale(1.1);
}

.about-link:hover .icon {
  opacity: 0.8;
}

/* Mobile responsive */
@media only screen and (max-width: 600px) {
  .icon {
    width: 40px;
  }
}

@media only screen and (max-width: 450px) {
  .icon {
    width: 35px;
  }
}
