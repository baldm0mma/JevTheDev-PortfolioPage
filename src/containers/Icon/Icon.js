import React from "react";

export const Icon = ({ source }) => {
  return (
    <a
      className="about-link"
      href={source.link}
      target="_blank"
      rel="noopener noreferrer"
      data-testid="icon-a-tag"
    >
      <img className="icon" src={source.imageSource} alt={source.altText} />
    </a>
  );
};
