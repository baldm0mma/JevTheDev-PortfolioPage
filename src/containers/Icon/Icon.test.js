import React from "react";
import { shallow } from "enzyme";
import { Icon } from "./Icon";

describe("Icon component", () => {
  let wrapper;
  let source;

  beforeEach(() => {
    source = {
      link: "www.resume.com",
      imageSource: "../assets/cv.png",
      altText: "Resume Icon"
    };
    wrapper = shallow(<Icon source={source} />);
  });

  it("should render without crashing", () => {
    expect(wrapper).toBeTruthy();
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should properly populate the dynamic attributes", () => {
    expect(wrapper.find("[href='www.resume.com']").length).toBe(1);
    expect(wrapper.find("[src='../assets/cv.png']").length).toBe(1);
    expect(wrapper.find("[alt='Resume Icon']").length).toBe(1);
  });
});
