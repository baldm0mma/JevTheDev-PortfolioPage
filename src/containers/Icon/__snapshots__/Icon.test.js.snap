// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component should match the snapshot 1`] = `
<a
  className="about-link"
  data-testid="icon-a-tag"
  href="www.resume.com"
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    alt="Resume Icon"
    className="icon"
    src="../assets/cv.png"
  />
</a>
`;
